dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -999,545 +799,431 @@ else if ( dataValueSet.getAttributeCategoryOptions() != null )
+    {
+        DataSet dataSet = dataValueSet.getDataSet() != null ? identifiableObjectManager.getObject( DataSet.class,
+            context.getDataSetIdScheme(), dataValueSet.getDataSet() ) : null;
+        CategoryOptionCombo outerAttrOptionCombo = null;
 
-                            auditBatchHandler.addObject( auditValue );
-                        }
-                    }
-                }
-            }
-            else
-            {
-                if ( strategy.isCreateAndUpdate() || strategy.isCreate() )
-                {
-                    if ( !internalValue.isNullValue() ) // Ignore null values
-                    {
-                        if ( existingValue != null && existingValue.isDeleted() )
-                        {
-                            importCount++;
-
-                            if ( !dryRun )
-                            {
-                                dataValueBatchHandler.updateObject( internalValue );
-
-                                if ( dataElement.isFileType() )
-                                {
-                                    FileResource fr = fileResourceService.getFileResource( internalValue.getValue() );
-
-                                    fr.setAssigned( true );
-
-                                    fileResourceService.updateFileResource( fr );
-                                }
-                            }
-                        }
-                        else
-                        {
-                            boolean added = false;
-
-                            if ( !dryRun )
-                            {
-                                added = dataValueBatchHandler.addObject( internalValue );
-
-                                if ( added && dataElement.isFileType() )
-                                {
-                                    FileResource fr = fileResourceService.getFileResource( internalValue.getValue() );
-
-                                    fr.setAssigned( true );
-
-                                    fileResourceService.updateFileResource( fr );
-                                }
-                            }
-
-                            if ( dryRun || added )
-                            {
-                                importCount++;
-                            }
-                        }
-                    }
-                }
-            }
+        if ( dataValueSet.getAttributeOptionCombo() != null )
+        {
+            outerAttrOptionCombo = context.getOptionComboMap().get(
+                trimToNull( dataValueSet.getAttributeOptionCombo() ),
+                context.getAttributeOptionComboCallable()
+                    .setId( trimToNull( dataValueSet.getAttributeOptionCombo() ) ) );
         }
-
-        dataValueBatchHandler.flush();
-
-        if ( !skipAudit )
+        else if ( dataValueSet.getAttributeCategoryOptions() != null && dataSet != null )
```

Comment:
The `&& dataSet != null` check was added a potential NPE was indicated by sonar.